docs/deploying-airbyte/on-azure-vm-cloud-shell.md

```diff
@@ -0,0 +1,138 @@
+# On Azure Virtual Machine - Cloud Shell
+
+{% hint style="info" %}
+The instructions have been tested on `Azure VM Linux (ubuntu 18.04)`
+{% endhint %}
+
+## Launch Azure Cloud Shell
+
+Launch cloud shell by going to [https://shell.azure.com/bash](https://shell.azure.com/bash)
+
+![](../.gitbook/assets/azure_shell_launch.png)
+
+## Create a new virtual machine
+
+#### Create resource group
+
+```bash
+# create resource group. yopu can change name and location
+rgName=airbyte-demo
+rgLocation=eastus
+az group create --name $rgName --location $rgLocation
+```
+![](../.gitbook/assets/azure_shell_create_rg.png)
+
+#### Create virtual machine
+
+```bash
+userName=byteuser
+vmName=airbyte
+dnsName=$(head -3 /dev/urandom | tr -dc a-z | cut -c -16)
+publicIp=$(az vm create --resource-group $rgName \
+  --name $vmName --image UbuntuLTS \
+  --admin-username $userName \
+  --public-ip-address-dns-name $dnsName \
+  --generate-ssh-keys --query 'publicIpAddress' -o json)
+echo $publicIp
+#get fqdns
+fqdns=$(az vm show -d -g $rgName -n $vmName --query fqdns -o tsv)
+echo $fqdns
+```
+
+This step will create virtual machine and add user account named `byteuser`. ``--generate-ssh-keys`` option will generate new ssh key and put it to default key location(~/.ssh)
+
+**_Note ``publicIpAddress`` from the output, you will need this addres to access airbyte_**
+
+![](../.gitbook/assets/azure_shell_create_vm.png)
+
+#### Open ports to access AirByte
+
+{% hint style="danger" %}
+For security reasons, we strongly recommend to not expose Airbyte on Internet available ports. Future versions will add support for SSL & Authentication.
+{% endhint %}
+
+```bash
+# open ports for airbyte
+az vm open-port --port 8000 --resource-group $rgName --name $vmName
+az vm open-port --port 8001 --resource-group $rgName --name $vmName
```

Comment:
just checking because i tried to run this and got the following error:
`(SecurityRuleConflict) Security rule open-port-8000 conflicts with rule open-port-8001. Rules cannot have the same Priority and Direction. To learn more, see aka.ms/nsgrules.`

Reply:
My bad, need to provide --priority for each port.